src/test/java/uk/gov/pay/api/it/PaymentsResourceITest.java

```diff
@@ -239,16 +238,41 @@ public void getPayment_Returns401_WhenUnauthorised() {
     }
 
     @Test
-    public void getPayment_InvalidPaymentId() {
-        String invalidPaymentId = "ds2af2afd3df112";
+    public void getPayment_returns404_whenConnectorRespondsWith404() throws IOException {
+
+        String paymentId = "ds2af2afd3df112";
         String errorMessage = "backend-error-message";
         publicAuthMock.mapBearerTokenToAccountId(BEARER_TOKEN, GATEWAY_ACCOUNT_ID);
-        connectorMock.respondChargeNotFound(GATEWAY_ACCOUNT_ID, invalidPaymentId, errorMessage);
+        connectorMock.respondChargeNotFound(GATEWAY_ACCOUNT_ID, paymentId, errorMessage);
 
-        getPaymentResponse(BEARER_TOKEN, invalidPaymentId)
+        InputStream body = getPaymentResponse(BEARER_TOKEN, paymentId)
                 .statusCode(404)
-                .contentType(JSON)
-                .body("message", is(errorMessage));
+                .contentType(JSON).extract()
+                .body().asInputStream();
+
+        JsonAssert.with(body)
+                .assertThat("$.*", hasSize(2))
+                .assertThat("$.code", is("P0200"))
+                .assertThat("$.description", is("Not found"));
+    }
+
+    @Test
+    public void getPayment_returns500_whenConnectorRespondsWithResponseOtherThan200Or404() throws IOException {
+
+        String paymentId = "ds2af2afd3df112";
+        String errorMessage = "backend-error-message";
+        publicAuthMock.mapBearerTokenToAccountId(BEARER_TOKEN, GATEWAY_ACCOUNT_ID);
+        connectorMock.respondWhenGetCharge(GATEWAY_ACCOUNT_ID, paymentId, errorMessage, SC_NOT_ACCEPTABLE);
+
+        InputStream body = getPaymentResponse(BEARER_TOKEN, paymentId)
+                .statusCode(500)
+                .contentType(JSON).extract()
+                .body().asInputStream();
+
+        JsonAssert.with(body)
+                .assertThat("$.*", hasSize(2))
+                .assertThat("$.code", is("P0298"))
+                .assertThat("$.description", is("Downstream system error"));
```

Comment:
The message is something still to show to them. Before was responding with whatever response came from Connector (and we don't want that)